flow/record/base.py

```diff
@@ -662,7 +688,56 @@ def DynamicDescriptor(name, fields):
     return RecordDescriptor(name, [("dynamic", field) for field in fields])
 
 
-def open_path(path, mode, clobber=True):
+def open_stream(fp: BinaryIO, mode: str) -> BinaryIO:
+    if not hasattr(fp, "peek"):
+        fp = Peekable(fp)
+
+    # We peek into the file at the maximum possible length we might need, which is the amount of bytes needed to
+    # determine whether a stream is a RECORDSTREAM or not.
+    peek_data = fp.peek(RECORDSTREAM_MAGIC_DEPTH)
+
+    # If the data stream is compressed, we wrap the file pointer in a reader that can decompress accordingly.
+    if peek_data[:2] == GZIP_MAGIC:
+        fp = gzip.GzipFile(fileobj=fp, mode=mode)
+    elif HAS_BZ2 and peek_data[:3] == BZ2_MAGIC:
+        fp = bz2.BZ2File(fp, mode=mode)
+    elif HAS_LZ4 and peek_data[:4] == LZ4_MAGIC:
+        fp = lz4.open(fp, mode=mode)
+    elif HAS_ZSTD and peek_data[:4] == ZSTD_MAGIC:
+        dctx = zstd.ZstdDecompressor()
+        fp = dctx.stream_reader(fp)
+
+    return fp
+
+
+def find_adapter_for_stream(fp: BinaryIO) -> tuple[Peekable, str]:
```

Comment:
```suggestion
def find_adapter_for_stream(fp: BinaryIO) -> tuple[BinaryIO, Optional[str]]:
```

Reply:
Could you elaborate on the change from `Peekable` to `BinaryIO`? In its current form, the `find_adapter_for_stream` function always wraps the given `fp` in a `Peekable` object before returning it.

Reply:
Does it matter for the return value though? Is functionality of `Peekable` ever required afterwards, or should it just behave as a `BinaryIO`?

Reply:
`io.BufferedReader` seems to be just fine so I'll make a commit that replaces the `Peekable` class with usage of `io.BufferedReader`. Thus you can ignore my previous comment.